src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,160 @@
+                || activityLog.getRoleWithoutMasquerade().contains(Const.ActivityLog.ROLE_INSTRUCTOR);
+    }
+
+    public String getUserIdentity() {
+        String googleId = activityLog.getGoogleId();
+        if (!googleId.contentEquals(Const.ActivityLog.AUTH_NOT_LOGIN)
+                && !googleId.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return googleId;
+        }
+
+        String email = activityLog.getEmail();
+        if (email != null && !email.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return email;
+        }
+
+        String name = activityLog.getName();
+        if (name != null && !name.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return name;
+        }
+        return "";
+    }
+
+    public boolean getHasUserEmail() {
+        return !Const.ActivityLog.UNKNOWN.contentEquals(activityLog.getEmail());
+    }
+
+    // --------------- 'is' fields to determine css class ---------------
+
+    public boolean getIsUserAdmin() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_ADMIN);
+    }
+
+    public boolean getIsUserInstructor() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_INSTRUCTOR);
+    }
+
+    public boolean getIsUserStudent() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_STUDENT);
+    }
+
+    public boolean getIsUserAuto() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_AUTO);
+    }
+
+    public boolean getIsUserUnregistered() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_UNREGISTERED);
+    }
+
+    public boolean getIsActionTimeTakenModerate() {
+        return activityLog.getTimeTaken() >= TIME_TAKEN_EXPECTED
+                && activityLog.getTimeTaken() <= TIME_TAKEN_MODERATE;
+    }
+
+    public boolean getIsActionTimeTakenSlow() {
+        return activityLog.getTimeTaken() > TIME_TAKEN_MODERATE;
+    }
+
+    public boolean getIsActionFailure() {
+        return activityLog.getAction().contains(Const.ACTION_RESULT_FAILURE);
+    }
+
+    public boolean getIsActionErrorReport() {
+        return activityLog.getAction().contains(Const.ACTION_RESULT_SYSTEM_ERROR_REPORT);
+    }
+
+    // --------------- Enhancement to the fields ---------------
+
+    public String getDisplayedActionUrl() {
+        return Url.addParamToUrl(activityLog.getUrl(),
+                                 Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+    }
+
+    public String getDisplayedLogTime() {
+        Calendar appCal = Calendar.getInstance(TimeZone.getTimeZone(Const.DEFAULT_TIMEZONE));
+        appCal.setTimeInMillis(activityLog.getTime());
+        appCal = TimeHelper.convertToUserTimeZone(appCal, Const.SystemParams.ADMIN_TIME_ZONE_DOUBLE);
+        return TimeHelper.calendarToString(appCal);
+    }
+
+    public String getDisplayedRole() {
+        return activityLog.getRole(); // will change later
```

Comment:
Normally I wouldn't allow this `// will change later`, but since #6531 is already there (I believe you'll do it there?), I'll let this one.